TweetMate/src/main/java/com/ybsystem/tweetmate/models/entities/twitter/TwitterStatus.java

```diff
@@ -147,19 +157,20 @@ public TwitterStatus(Status status) {
         this.convertedRelativeTime = createRelativeTime(status);
         this.convertedAbsoluteTime = createAbsoluteTime(status);
         this.convertedVia = createVia(status);
+        this.imageUrls = createImageUrls(status);
         this.isMyTweet = status.getUser().getId() == TweetMateApp.getMyUser().getId();
         this.isPublic = isMyTweet || !status.getUser().isProtected();
         this.isTalk = status.getInReplyToStatusId() != -1;
         this.isDetail = false;
-        this.isMediaTweet = !status.isRetweet() && status.getMediaEntities().length != 0;
+        this.isTwiMedia = (imageUrls.size() > 0 && mediaEntities.length > 0);
+        this.isThirdMedia = (imageUrls.size() > 0  && mediaEntities.length == 0);
+        this.isVideo = (isTwiMedia && !mediaEntities[0].getType().equals("photo"));
     }
 
     private static String createTweetText(Status status) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3776)

Reply:
Pending